<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
<head>
<meta http-equiv="Content-Type" content="application/xhtml+xml; charset=UTF-8" />
<title>GenomeTools - manual page for GT-REPFIND(1)</title>
<link rel="stylesheet" type="text/css" href="style.css">
<link rel="stylesheet" href="..//style.css" type="text/css" />
</head>
<body>
<div id="menu">
<ul>
<li><a href="../index.html">Overview</a></li>
<li><a href="https://github.com/genometools/genometools/releases">Releases</a></li>
<li><a href="pub/">Archive</a></li>
<li><a href="https://github.com/genometools/genometools">Browse source</a></li>
<li><a href="http://github.com/genometools/genometools/issues/">Issue tracker</a></li>
<li><a href="../documentation.html">Documentation</a></li>
  <ul class="submenu">
    <li><a id="current" href="../tools.html">Tools</a></li>
    <li><a href="../manuals.html">Manuals</a></li>
    <li><a href="../libgenometools.html">C API</a></li>
    <li><a href="../docs.html"><tt>gtscript</tt> docs</a></li>
    <li><a href="../contract.html">Development Contract</a></li>
    <li><a href="../contribute.html">Contribute</a></li>
  </ul>
<li><a href="../annotationsketch.html"><tt>AnnotationSketch</tt></a></li>
<li><a href="../cgi-bin/gff3validator.cgi">GFF3 validator</a></li>
<li><a href="../license.html">License</a></li>
</ul>
</div>
<div id="main">
<div class="sect1">
<h2 id="_name">NAME</h2>
<div class="sectionbody">
<div class="paragraph"><p>gt-repfind - Compute maximal exact matches (and more).</p></div>
</div>
</div>
<div class="sect1">
<h2 id="_synopsis">SYNOPSIS</h2>
<div class="sectionbody">
<div class="paragraph"><p><strong>gt repfind</strong> [options] -ii indexname</p></div>
</div>
</div>
<div class="sect1">
<h2 id="_description">DESCRIPTION</h2>
<div class="sectionbody">
<div class="dlist"><dl>
<dt class="hdlist1">
<strong>-l</strong> [<em>value</em>]
</dt>
<dd>
<p>
Specify minimum length of matches (default: 0)
</p>
</dd>
<dt class="hdlist1">
<strong>-f</strong> [<em>yes|no</em>]
</dt>
<dd>
<p>
Compute forward matches (default: yes)
</p>
</dd>
<dt class="hdlist1">
<strong>-r</strong> [<em>yes|no</em>]
</dt>
<dd>
<p>
Compute reverse matches (default: no)
</p>
</dd>
<dt class="hdlist1">
<strong>-p</strong> [<em>yes|no</em>]
</dt>
<dd>
<p>
Compute matches on reverse strand (default: no)
</p>
</dd>
<dt class="hdlist1">
<strong>-seedlength</strong> [<em>value</em>]
</dt>
<dd>
<p>
Specify minimum length of seed (default: 0)
</p>
</dd>
<dt class="hdlist1">
<strong>-maxfreq</strong> [<em>value</em>]
</dt>
<dd>
<p>
Specify maximal frequency of maximal exact matches in reference sequence (default: 0)
</p>
</dd>
<dt class="hdlist1">
<strong>-extendxdrop</strong> [<em>value</em>]
</dt>
<dd>
<p>
Extend seed to both sides using xdrop algorithm, optional parameter specifies sensitivity (default: 97)
</p>
</dd>
<dt class="hdlist1">
<strong>-xdropbelow</strong> [<em>value</em>]
</dt>
<dd>
<p>
Specify xdrop cutoff score (argument 0 means undefined). If undefined an optimal value is determined automatically depending on the error rate
</p>
</dd>
<dt class="hdlist1">
<strong>-extendgreedy</strong> [<em>value</em>]
</dt>
<dd>
<p>
Extend seed to both sides using greedy algorithm with trimming of waves, optional parameter specifies sensitivity (default: 97)
</p>
</dd>
<dt class="hdlist1">
<strong>-minidentity</strong> [<em>value</em>]
</dt>
<dd>
<p>
Specify minimum identity of matches
as integer in the range from 70 to 99 (for xdrop and greedy extension) (default: 80)
</p>
</dd>
<dt class="hdlist1">
<strong>-ii</strong> [<em>string</em>]
</dt>
<dd>
<p>
Specify input index (default: undefined)
</p>
</dd>
<dt class="hdlist1">
<strong>-outfmt</strong> 
</dt>
<dd>
<p>
specify what information about the matches to display
alignment:        show alignment (possibly followed by
                  =&lt;number&gt; to specify width of alignment
                  columns, default is )
seed_in_algn:     mark the seed in alignment
polinfo:          add polishing information for shown
                  alignment
seed:             abbreviation for seed.len seed.s seed.q
failed_seed:      show the coordinates of a seed extension,
                  which does not satisfy the filter conditions
fstperquery:      output only the first found match per query
tabsep:           separate columns by tabulator, instead of
                  space as default
blast:            output matches in blast format 7 (tabular with
                  comment lines; instead of gap opens, indels are
                  displayed)
gfa2:             output matches in gfa2 format
custom:           output matches in custom format, i.e. no
                  columns are pre-defined; all columns have to be
                  specified by the user
cigar:            display cigar string representing alignment
                  (no distinction between match and mismatch)
cigarX:           display cigar string representing alignment
                  (distinction between match (=) and mismatch
                  (X))
trace:            display trace, i.e. a compact representation
                  of an alignment (possibly followed by =&lt;delta&gt;)
                  to specify the delta-value; default value of
                  delta is 50
dtrace:           display trace as differences; like trace, but
                  instead of an absolute value x, report the
                  difference delta-x. This leads to smaller
                  numbers and thus a more compact representation
s.len:            display length of match on subject sequence
s.seqnum:         display sequence number of subject sequence
subject id:       display sequence description of subject
                  sequence
s.start:          display start position of match on subject
                  sequence
s.end:            display end position of match on subject
                  sequence
strand:           display strand of match using symbols F
                  (forward) and P (reverse complement)
q.len:            display length of match on query sequence
q.seqnum:         display sequence number of query sequence
query id:         display sequence description of query
                  sequence
q.start:          display start position of match on query
                  sequence
q.end:            display end position of match on query
                  sequence
alignment length: display length of alignment
mismatches:       display number of mismatches in alignment
indels:           display number of indels in alignment
gap opens:        display number of indels in alignment
score:            display score of match
editdist:         display unit edit distance
identity:         display percent identity of match
seed.len:         display length seed of the match
seed.s:           display start position of seed in subject
seed.q:           display start position of seed in query
s.seqlen:         display length of subject sequence in which
                  match occurs
q.seqlen:         display length of query sequence in which
                  match occurs
evalue:           display evalue
bit score:        display bit score
</p>
</dd>
<dt class="hdlist1">
<strong>-evalue</strong> [<em>value</em>]
</dt>
<dd>
<p>
switch on evalue filtering of matches (optional argument specifies evalue threshold) (default: 10.000000)
</p>
</dd>
<dt class="hdlist1">
<strong>-scan</strong> [<em>yes|no</em>]
</dt>
<dd>
<p>
scan index rather than map it to main memory (default: no)
</p>
</dd>
<dt class="hdlist1">
<strong>-v</strong> [<em>yes|no</em>]
</dt>
<dd>
<p>
be verbose (default: no)
</p>
</dd>
<dt class="hdlist1">
<strong>-help</strong> 
</dt>
<dd>
<p>
display help and exit
</p>
</dd>
<dt class="hdlist1">
<strong>-version</strong> 
</dt>
<dd>
<p>
display version information and exit
</p>
</dd>
</dl></div>
</div>
</div>
<div class="sect1">
<h2 id="_reporting_bugs">REPORTING BUGS</h2>
<div class="sectionbody">
<div class="paragraph"><p>Report bugs to &lt;<a href="mailto:kurtz@zbh.uni-hamburg.de">kurtz@zbh.uni-hamburg.de</a>&gt;.</p></div>
</div>
</div>
<div id="footer">
Copyright &copy; 2007-2023 The <i>GenomeTools</i> authors.
</div>
</div>
</body>
</html>
